src/index.js

```diff
@@ -217,28 +217,35 @@ function UseWalletProvider({
     [chainId, connectorsInitsOrConfigs]
   )
 
-  const deactivate = useCallback(async () => {
+  const reset = useCallback(() => {
     if (web3ReactContext.active) {
-      await web3ReactContext.deactivate()
+      web3ReactContext.deactivate()
     }
-    setActivating(null)
-    setActivated(null)
+    setConnector(null)
+    setError(null)
+    setStatus('disconnected')
   }, [web3ReactContext])
 
-  const activate = useCallback(
+  const connect = useCallback(
     async (connectorId = 'injected') => {
       // Prevent race conditions between connections by using an external ID.
       const id = ++activationId.current
 
-      deactivate()
+      reset()
 
       // Check if another connection has happened right after deactivate().
       if (id !== activationId.current) {
         return
       }
 
+      // If no connection happene, we're in the right context, and we can safely update
```

Comment:
```suggestion
      // If no connection happens, we're in the right context and can safely update
```
Do you think this reads better?

Reply:
Definitely!